### PR TITLE
Message queue for waiting player

### PR DESCRIPTION
**Pull request by gacikjakub**
Generally there is mechanism which allow us to simple initialize game and safe transfer data to waiting player which keeps asking backend about opponent shot result.

### CODE REVIEW

**Review by krzysieksr — APPROVED** [review]

**Comment by gacikjakub** [comment]
Build 42 Successful
Deployed under: message-queue-for-waiting-player.battleships.kuba005.nazwa.pl

File: backend/src/main/java/pl/krkteam/battleships/opponent_shot_result_keeping/ShotResultQueueHolder.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+    private void initializeQueueHolder(Player player1, Player player2) {
+        if (player1 == null || player2 == null) {
+            throw new IllegalArgumentException("Player cannot be null");
+        }
+        playerShotResultQueueMap.put(player1, new ShotResultQueue());
+        playerShotResultQueueMap.put(player2, new ShotResultQueue());
+    }
+
+    public ShotResultQueueHolder() {}
```
Comment (xKoem) [inline backend/src/main/java/pl/krkteam/battleships/opponent_shot_result_keeping/ShotResultQueueHolder.java]: reordering.. Constructors first

File: backend/src/main/java/pl/krkteam/battleships/opponent_shot_result_keeping/ShotResultQueueHolder.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    public ShotResultQueueHolder(Player player1, Player player2) {
+        initializeQueueHolder(player1, player2);
+    }
+
+    public ShotResultQueue getShotResultQueue(Player player) {
+        if (player == null) {
+            throw new IllegalArgumentException("Player cannot be null");
+        }
+        if (!playerShotResultQueueMap.containsKey(player)) {
+            throw new NoSuchElementException("Given player does not belong to this holder");
```
Comment (xKoem) [inline backend/src/main/java/pl/krkteam/battleships/opponent_shot_result_keeping/ShotResultQueueHolder.java:42]: Maybe add own exceptions

File: backend/src/main/test/java/pl/krkteam/battleships/common/domain/GameBoardHolderTest.java [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    GameBoard gameBoard2;
+
+    @BeforeClass
+    @BeforeMethod
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test(expectedExceptions = IllegalArgumentException.class)
+    public void shouldThrowExceptionWhenGivenPlayerToAddGameBoardIsNull() {
```
Comment (xKoem) [inline backend/src/main/test/java/pl/krkteam/battleships/common/domain/GameBoardHolderTest.java:39]: shouldThrowException_whenGivenPlayerToAddGameBoardIsNull - could have

**Review by xKoem — CHANGES_REQUESTED** [review]

**Review by xKoem — APPROVED** [review]

File: backend/src/main/java/pl/krkteam/battleships/common/domain/GameBoardHolder.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        if (playerGameBoardMap.containsKey(player)) {
+            return false;
+        }
+        playerGameBoardMap.put(player, gameBoard);
+        return true;
+    }
+
+    public GameBoard getGameBoard(Player player) {
+        if (player == null) {
+            throw new IllegalArgumentException("Player cannot be null");
```
Comment (AdrianHarenczyk) [inline backend/src/main/java/pl/krkteam/battleships/common/domain/GameBoardHolder.java:29]: Could be refactored to separate method. It is code duplication.

**Review by AdrianHarenczyk — APPROVED** [review]
Mainly ok :)

**Pull request merged**